test/libsolidity/SemanticTest.cpp

```diff
@@ -239,12 +237,51 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 	return TestResult::Success;
 }
 
-void SemanticTest::printSource(ostream& _stream, string const& _linePrefix, bool) const
+void SemanticTest::printSource(ostream& _stream, string const& _linePrefix, bool _formatted) const
 {
-	stringstream stream(m_source);
-	string line;
-	while (getline(stream, line))
-		_stream << _linePrefix << line << endl;
+	if (m_sources.sources.empty())
+		return;
+
+	bool outputSourceNames = true;
+	if (m_sources.sources.size() == 1 && m_sources.sources.begin()->first.empty())
+		outputSourceNames = false;
+
+	if (_formatted)
+		for (auto const& [name, source]: m_sources.sources)
```

Comment:
Can you swap the `for` and the `if (_formatted)`?

Reply:
I applied this change to the original code in `CommonSyntaxTest.cpp` as well